_posts/2018-12-05-exploit-model.md

```diff
@@ -0,0 +1,195 @@
+---
+layout: post
+title: Discussing Exploitation and Priv Escalation - Analysis of gVisor exploit
+date: 2018-12-05
+author: lumjjb
+description: Discussing Exploitation and Priv Escalation - Analysis of gVisor exploit
+---
+
+In this blog post, we take a look at the work that Max Justicz wrote about in his post ["Privilege Escalation in gVisor, Google's Container Sandbox"](https://justi.cz/security/2018/11/14/gVisor-lpe.html). We discuss the threat model of gVisor and what the exploit provides and the differences between the LibOS security model between gVisor and Nabla containers. Finally, we explore some other threats.
+
+## Exploitation 102: Attacker capabilities
+
+To understand a few concepts that we would like to talk about, let us go into a quick primer on how exploits are written. There are multiple reasons why an attacker would like to exploit an application. The most common reasons are that they want to leak some data from the application, or they want to be able to execute arbitrary code in the application (code exec).
+
+Let's assume that we are exploiting an application (this could be something like redis or nginx) and we want to achieve arbitrary code execution (code exec).
+
+Starting with our goal, we want to describe how an attacker derives an exploit - arbitrary code execution. How do we get arbitrary code execution? One way of doing that is by hijacking the control flow of the application. This can be done via various methods, like overwriting function pointers, overwriting an entry in the `.got` (Global Offset Table entries contain function pointers of dynamically linked libraries), or taking advantage of architecture specific behaviors (like overwriting the return address in the stack used by the `ret` instruction), etc.
+
+The main way of figuring out how to do this is by reverse engineering the application binary and finding bugs in the application. For example, a failure to check bounds may lead to a leak of stack data or heap data ([Heartbleed](http://heartbleed.com/)). And a use after free bug in memory allocation may be able to provide the ability to write to an arbitrary memory address. 
```

Comment:
"use after free bug" --> "use-after-free bug"  (i think you connect the words with hyphens if the phrase acts as an adjective)

Reply:
Yup! Made the change throughout the doc.